File: spring-boot/grpc/server/src/main/java/no/acntech/sandbox/service/GreetingService.java
package no.acntech.sandbox.service;

import io.grpc.stub.StreamObserver;
import net.devh.boot.grpc.server.service.GrpcService;
import no.acntech.sandbox.model.GreetingGrpc;
import no.acntech.sandbox.model.HelloRequest;
import no.acntech.sandbox.model.HelloResponse;

@GrpcService
public class GreetingService extends GreetingGrpc.GreetingImplBase {

    @Override
    public void sayHello(HelloRequest request, StreamObserver<HelloResponse> responseObserver) {
        var response = HelloResponse.newBuilder()
                .setMessage("Hello " + request.getName() + "!")
                .build();
        responseObserver.onNext(response);
        responseObserver.onCompleted();
    }
}

File: spring-boot-kotlin/grpc/server/src/main/kotlin/no/acntech/sandbox/service/GreetingService.kt
package no.acntech.sandbox.service

import io.grpc.stub.StreamObserver
import net.devh.boot.grpc.server.service.GrpcService
import no.acntech.sandbox.model.GreetingGrpc
import no.acntech.sandbox.model.HelloRequest
import no.acntech.sandbox.model.HelloResponse

@GrpcService
class GreetingService : GreetingGrpc.GreetingImplBase() {

    override fun sayHello(request: HelloRequest, responseObserver: StreamObserver<HelloResponse?>) {
        val response = HelloResponse.newBuilder()
                .setMessage("Hello ${request.name}!")
                .build()
        responseObserver.onNext(response)
        responseObserver.onCompleted()
    }
}